src/renderer/kube-object.store.ts

```diff
@@ -106,17 +106,18 @@ export abstract class KubeObjectStore<T extends KubeObject = any> extends ItemSt
   }
 
   @action
-  async loadAll({ namespaces: contextNamespaces }: { namespaces?: string[] } = {}) {
+  async loadAll(namespaces: string[] = []): Promise<void> {
     this.isLoading = true;
 
     try {
-      if (!contextNamespaces) {
+      if (!namespaces.length) {
         const { namespaceStore } = await import("./components/+namespaces/namespace.store");
 
-        contextNamespaces = namespaceStore.getContextNamespaces();
+        // load all available namespaces by default
+        namespaces.push(...namespaceStore.allowedNamespaces);
```

Comment:
First of all this doesn't need to be a `push(...)`, it could be an assign.

Second of all, why is `NamespaceStore` getting a "get-er" that merely delegates to `getHostedCluster()`? I don't think we should have that (it doesn't make sense for `NamespaceStore` to have that function) and it adds this circular dependency.

Lastly, I don't think this is how we should be "hiding getHostedCluster".

Reply:
Yes, `NamespaceStore` will be refactored similarly as in `KubeWatchApi`:

```
  @observable.ref private cluster: Cluster;
  @observable.ref private namespaces: string[] = [];
```

Anyway it's not critical for now and could be improved in follow-up PR(s).